Lib/test/test_socket.py

```diff
@@ -5177,6 +5177,24 @@ def test_create_connection(self):
         expected_errnos = socket_helper.get_socket_conn_refused_errs()
         self.assertIn(cm.exception.errno, expected_errnos)
 
+    def test_create_connection_all_errors(self):
+        port = socket_helper.find_unused_port()
+        try:
+            socket.create_connection((HOST, port), all_errors=True)
+        except* OSError as e:
```

Comment:
```suggestion
        except ExceptionGroup as e:
```

(This makes sure that an ExceptionGroup was actually raised, rather than except* wrapping an OSError for us).